src/dataset/create_dataset.py

```diff
@@ -19,240 +20,141 @@
-        elif 'book_title' in article and 'title' not in article and article['book_title']: 
-            title = article['book_title']
-        elif 'title' in article and article['title']: 
-            title = article['title']
-        title = title.strip('\n').replace('\n', ' ')[:100]
-
-        # Get author
-        if 'author' in article and 'authors' in article and article['author']: author = article['author']
-        elif 'authors' in article and article['authors']: author = article['authors']
-        elif 'author' in article and article['author']: author = article['author']
-        if type(author) == str: author = get_authors_list(author)
-        if type(author) == list: author = ', '.join(author)
-        author = author.strip('\n').replace('\n', ' ')[:100]
-
-        # Get date published
-        if 'date_published' in article and article['date_published'] and len(article['date_published']) >= 10: date_published = article['date_published'][:10]
-        elif 'published' in article and article['published'] and len(article['published']) >= 16: date_published = article['published'][:16]
-        else: date_published = None
-        if date_published is not None:
-            date_published = standardize_date(date_published)
-            
-        # Get URL
-        if 'link' in article and article['link']: url = article['link']
-        elif 'url' in article and article['url']: url = article['url']
-        elif 'doi' in article and article['doi']: url = article['doi']
-        else: url = None
-            
-        # Get tags
-        if 'tags' in article and article['tags']:
-            if type(article['tags']) == list: tags = ', '.join([val['term'] for val in article['tags']])
-            elif type(article['tags']) == str: tags = article['tags']
-            else: tags = None
+                self.entries_per_source_count[source] = 0
         
-        # Get text
-        if 'text' in article and article['text']: text = article['text']
-        else:
-            raise MissingDataException(f"Entry has no text.")
+    def contains_required_metadata(self, entry: Dict[str, Any]):
+        metadata_types = {
+            'id': str,
+            'source': str,
+            'title': str,
+            'url': str,
+            'date_published': str,
+            'authors': list, # unsure as of yet if this is a string or a list. TODO it is subject to change
+#           'summary': list  # see previous comment
+        }
+        required_metadata_keys = ['id', 'source', 'title', 'text']
+
+        # Check that the 8 primary metadata keys all have the correct type
+        for key, key_type in metadata_types.items():
+            if type(entry[key]) != key_type:
+                raise MissingDataException(f"Entry {entry['id']} has key {key} of type {type(entry[key])} when it should be {key_type}.")
+
+        # Check that the 4 required metadata keys are non-empty
+        for key in required_metadata_keys:
```

Comment:
just do `for key in metadata_types` - all the keys should be present

Reply:
I'm checking whether they're non-empty though, meaning a summary of [] triggers the error currently, which is common. I can instead check that 'key' in entry.keys(), which would check if 'summary' is a key of the entry even if it's an empty list or whatever